client/signup/steps/design-picker/index.jsx

```diff
@@ -79,6 +80,11 @@ class DesignPickerStep extends Component {
 		// `/start` and `/new` onboarding flows. Or perhaps fetching should be done within the <DesignPicker>
 		// component itself. The `/new` environment needs helpers for making authenticated requests to
 		// the theme API before we can do this.
+
+		if ( 'do-it-for-me' === this.props.flowName ) {
+			return DoItForMeThemes;
+		}
+
```

Comment:
I think it's better to use a step PROP to achieve this rather than touch these step internals. We can inject the themes if available and use the recommended Themes when any themes are not injected. This approach further conforms to the theme object shape we follow making it easy to do any future changes.

https://github.com/Automattic/wp-calypso/blob/1ea30608f4c71ad21a885ac3c35cd4cdc7fe0011/client/signup/steps/design-picker/index.jsx#L254

Reply:
The structure required is quite simple as shown below.
https://github.com/Automattic/wp-calypso/blob/1ea30608f4c71ad21a885ac3c35cd4cdc7fe0011/client/signup/steps/design-picker/index.jsx#L93

And the structure required to feed to the design picker component will be generated by the `getDesigns` internal function (which we do not touch).

https://github.com/Automattic/wp-calypso/blob/1ea30608f4c71ad21a885ac3c35cd4cdc7fe0011/client/signup/steps/design-picker/index.jsx#L88-L104

We just have to make sure our that our `(taxonomies.theme_subject : name)` has a strict (1:N) (No intersection) mapping.

Reply:
Thanks for the great feedback! I've updated the PR, please let me know if the new changes make sense.